### PR TITLE
V2 Reconfig - Project details

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

This PR creates the base reconfig modal, but only implements the ability to reconfigure 'Project details' (metadata stuff). 

As well as obvious new v2 Reconfigure components:
- v2 `HasPermission` hook used to conditionally show the reconfigure modal trigger
- v2 `OwnerOfProject` hook
- v2 `EditV2ProjectDetailsTx` tx

## Screenshots or screen recordings

https://user-images.githubusercontent.com/96150256/159147683-557eb6fd-2964-4a26-a06f-eb9ed3189cfc.mp4

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](../CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](../CONTRIBUTING.md#browser-support).

### CODE REVIEW

**Comment by aeolianeth** [comment]
@johnnyd-eth looks like theres some slight regressions in the UI. Mind taking a look? Also, do we want to have the ⚙️  icon in the header to trigger this modal like we do currently?

| `main` | this PR |
| --- | --- |
| Screen Shot 2022-03-21 at 11 33 47 am | Screen Shot 2022-03-21 at 11 33 50 am |

**Comment by aeolianeth** [comment]
**issue**: my project details don't appear to be pre-filling in the form:

Screen Shot 2022-03-21 at 11 38 02 am

**Comment by aeolianeth** [comment]
**issue** project handles don't exist in V2, can we remove that field from the form? We can pass `hideProjectHandle` prop to the form component to achieve this.

**Comment by johnnyd-eth** [comment]
@aeolianeth Think you might be behind my latest

File: src/components/v2/V2Project/V2ProjectReconfigureModal/V2ReconfigureModalTrigger.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    useState<boolean>(false)
+
+  function handleModalOpen() {
+    localStoreRef.current = createStore()
+    setReconfigureModalVisible(true)
+  }
+
+  return (
+    <div style={{ textAlign: 'right' }}>
+      {/* {fundingDuration?.gt(0) ? (
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2ProjectReconfigureModal/V2ReconfigureModalTrigger.tsx:38]: @johnnyd-eth in general lets try to minimize commented code in the codebase. Find to leave this one though

**Pull request merged**